ktor-server/ktor-server-core/jvm/src/io/ktor/response/ApplicationResponseFunctions.kt

```diff
@@ -12,22 +12,53 @@ import io.ktor.http.content.*
 import io.ktor.util.*
 import io.ktor.utils.io.*
 import java.io.*
+import kotlin.reflect.*
 
 /**
  * Sends a [message] as a response
  */
-@Suppress("NOTHING_TO_INLINE")
+@OptIn(ExperimentalStdlibApi::class)
+@JvmName("respondWithType")
+public suspend inline fun <reified T> ApplicationCall.respond(message: T) {
+    try {
+        // We need to wrap getting type in try catch because of https://youtrack.jetbrains.com/issue/KT-42913
+        attributes.put(ResponseTypeAttributeKey, typeOf<T>())
+    } catch (_: Throwable) {
+    }
+    response.pipeline.execute(this, message as Any)
```

Comment:
On the other hand, a `null` value will crash here with KNPE. So we need to fix and allow this or we need to add an upper bound for type `T`.

Reply:
limited type to `T : Any`